src/mlpack/tests/activation_functions_test.cpp

```diff
@@ -406,6 +407,55 @@ BOOST_AUTO_TEST_CASE(SELUFunctionDerivativeTest)
       selu.Lambda() * selu.Alpha() - arma::mean(activations))), 10e-4);
 }
 
+/*
+ * Implementation of the CELU activation function test. The function is
+ * implemented as CELU layer in the file celu.hpp
+ *
+ * @param input Input data used for evaluating the CELU activation function.
+ * @param target Target data used to evaluate the CELU activation.
+ */
+void CheckCELUActivationCorrect(const arma::colvec input,
+                                     const arma::colvec target)
+{
+  // Initialize CELU object with alpha = 1.0.
+  CELU<> lrf(1.0);
+
+  // Test the activation function using the entire vector as input.
+  arma::colvec activations;
+  lrf.Forward(std::move(input), std::move(activations));
+  for (size_t i = 0; i < activations.n_elem; i++)
+  {
+    BOOST_REQUIRE_CLOSE(activations.at(i), target.at(i), 1e-3);
+  }
+}
+
+/*
+ * Implementation of the CELU activation function derivative test. The function
+ * is implemented as CELU layer in the file celu.hpp
+ *
+ * @param input Input data used for evaluating the CELU activation function.
+ * @param target Target data used to evaluate the CELU activation.
+ */
+void CheckCELUDerivativeCorrect(const arma::colvec input,
```

Comment:
Ohh I understand why you did this, because CheckDerivative correct doesn't pass a value for alpha, I think it would be good idea then to assign a default value to alpha in constructor of CELU.

Reply:
But, @kartikdutt18 @himanshupathak21061998, the CELU activation function has been implemented as a layer and all the activation functions which have been implemented as a layer, for eg:- ELU, Hard TanH, etc, have their own functions in 'activation_functions_test.cpp'. Correct me if I am missing something.

Reply:
Yes you are right however I felt that by using a default alpha value you could reduce the code you write. You can keep it the way it is if needed.

Reply:
May be it's fine but member will give more insight about it.

Reply:
Sure, @gaurav-singh1998, You can resolve this conversation if needed.

Reply:
Since the layer methods aren't static, we can't reuse `CheckDerivative`, but maybe it would be great to provide another implementation for layers I think that would be useful in other places as well. Let me know what you think.

Reply:
Hi @zoq, I think we should to this. So, should I go ahead and try to implement it? If I get it correct will open another PR for it.

Reply:
Agreed, that will make it look a lot cleaner / remove similar code.

Reply:
Hi, @zoq @kartikdutt18 after inspecting **activation_functions_test.cpp** I don't think we can provide a separate implementation for activations functions which are implemented are as layers because,
- Tests for some function use `BOOST_REQUIRE_CLOSE()` for eg:- **HardTanH**, **ELU**, etc while some function use `BOOST_REQUIRE_EQUAL()` for eg:- **PReLU**.
- Test for the activation function **SELU** check `whether the mean and variance remain invariant after passing normalized inputs through the function` and `whether the mean and variance vary significantly after passing unnormalized inputs through the function`.

Above are some of the reasons due to which I think there cannot be a generalized implementation for activation functions which are implemented as layers. Let me know if I am wrong somewhere. Thanks